controllers/awsmachine_controller_test.go

```diff
@@ -342,31 +358,148 @@ var _ = Describe("AWSMachineReconciler", func() {
 					Expect(ms.AWSMachine.Status.FailureMessage).To(PointTo(Equal("EC2 instance state \"terminated\" is unexpected")))
 				})
+			})
+		})
+	})
+
+	Context("secrets management lifecycle", func() {
+		var instance *infrav1.Instance
+		var buf *bytes.Buffer
+
+		When("creating EC2 instances", func() {
+			BeforeEach(func() {
+				buf = new(bytes.Buffer)
+				klog.SetOutput(buf)
+				ec2Svc.EXPECT().GetRunningInstanceByTags(gomock.Any()).Return(nil, nil).AnyTimes()
+				ec2Svc.EXPECT().CreateInstance(gomock.Any(), gomock.Any()).Return(instance, nil).AnyTimes()
+			})
+
+			When("the machine is a control plane one", func() {
+				It("should leverage AWS Secrets Manager", func() {
+					ms.Machine.ObjectMeta.Labels = map[string]string{
+						clusterv1.MachineControlPlaneLabelName: "",
+					}
+					arn := pointer.StringPtr("testARN")
+					secretSvc.EXPECT().Create(gomock.Any(), gomock.Any()).Return(arn, nil).Times(1)
+					_, _ = reconciler.reconcileNormal(context.Background(), ms, cs)
+					Expect(ms.AWSMachine.Spec.CloudInit.Encrypted).To(Equal(true))
+					Expect(ms.AWSMachine.Spec.CloudInit.SecretARN).To(Equal(arn))
+				})
+			})
+
+			When("the machine isn't a control plane one", func() {
+				It("should not leverage AWS Secrets Manager", func() {
+					ms.Machine.ObjectMeta.Labels = map[string]string{}
+					secretSvc.EXPECT().Create(gomock.Any(), gomock.Any()).Times(0)
+					_, _ = reconciler.reconcileNormal(context.Background(), ms, cs)
+					Expect(ms.AWSMachine.Spec.CloudInit).To(BeNil())
+				})
+			})
+		})
+
+		When("there's a node ref and a secret ARN", func() {
+			BeforeEach(func() {
+
+				buf = new(bytes.Buffer)
+				klog.SetOutput(buf)
+
+				instance = &infrav1.Instance{
+					ID: "myMachine",
+				}
+
+				ms.Machine.Status.NodeRef = &corev1.ObjectReference{
+					Kind:       "Node",
+					Name:       "myMachine",
+					APIVersion: "v1",
+				}
 
+				ms.AWSMachine.Spec.CloudInit = &infrav1.CloudInit{
+					SecretARN: pointer.StringPtr("secret"),
+				}
+				ec2Svc.EXPECT().GetRunningInstanceByTags(gomock.Any()).Return(nil, nil).AnyTimes()
+				ec2Svc.EXPECT().CreateInstance(gomock.Any(), gomock.Any()).Return(instance, nil).AnyTimes()
+			})
+
+			It("should delete the secret if the instance is running", func() {
+				instance.State = infrav1.InstanceStateRunning
+				ec2Svc.EXPECT().GetInstanceSecurityGroups(gomock.Any()).
+					Return(map[string][]string{"eid": {}}, nil).Times(1)
+				ec2Svc.EXPECT().GetCoreSecurityGroups(gomock.Any()).Return([]string{}, nil).Times(1)
+				secretSvc.EXPECT().Delete(gomock.Any()).Return(nil).Times(1)
+				_, _ = reconciler.reconcileNormal(context.Background(), ms, cs)
+			})
+
+			It("should delete the secret if the instance is terminated", func() {
+				instance.State = infrav1.InstanceStateTerminated
+				secretSvc.EXPECT().Delete(gomock.Any()).Return(nil).Times(1)
+				_, _ = reconciler.reconcileNormal(context.Background(), ms, cs)
+			})
```

Comment:
If you add the delete call in the failure short-circuit, add a test case for a failed machine (should delete secret)